skbio/stats/composition.py

```diff
@@ -608,10 +612,274 @@ def centralize(mat):
+       Lower bound for the proportion of differiential features
+       (default=0.1)
+    significance_test: function, optional
+       A statistical signficance function to test for
+       signficance between classes.
+       The default is scipy.stats.ttest_ind
+
+    Returns:
+    --------
+    pd.DataFrame
+        A table of features, their W-statistics and whether
+        the null hypothesis is rejected
+
+    References
+    ----------
+    ..[1] S. Mandal, "Analysis of composition of microbiomes:
+          a novel method for studying microbial composition",
+          Microbial Ecology in Health & Disease, (2015), 26
+
+    Examples
+    --------
+    First import all of the necessary modules
+
+    >>> from skbio.stats.composition import ancom
+    >>> import pandas as pd
+
+    Now lets load in a pandas dataframe with sample and feature ids
+    for our data matrix.
+    >>> table = pd.DataFrame(
+    ...     [[10., 11., 10., 10., 10., 10., 10.],
+    ...      [10.5, 11.5, 10.5, 10.5, 10.5, 10.5, 10.5],
+    ...      [10., 11., 10., 10., 10., 10., 10.],
+    ...      [20., 21., 10., 10., 10., 10., 10.],
+    ...      [20.5, 21.5, 10.5, 10.5, 10.5, 10.5, 10.5],
+    ...      [20.3, 21.3, 10.2, 10.3, 10.1, 10.6, 10.4]],
+    ...     index=['s1','s2','s3','s4','s5','s6'],
+    ...     columns=['b1','b2','b3','b4','b5','b6','b7'])
+
+    Then create a create a category vector.  In this scenerio, there
+    are only two classes, so the first three samples fall under the first
+    class while the last three samples fall under the last class
+    >>> cats = pd.Series([0, 0, 0, 1, 1, 1],
+    ...                  index=['s1','s2','s3','s4','s5','s6'])
+
+    Now run ancom and see if there are any features that have any
+    significant differences
+    >>> results = ancom(table, cats)
+    >>> results['W']
+    b1    6
+    b2    6
+    b3    2
+    b4    2
+    b5    2
+    b6    2
+    b7    2
+    Name: W, dtype: int64
+    >>> results['reject']
+    b1     True
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    Name: reject, dtype: bool
+    """
+    if len(matrix) != len(categories):
+        raise ValueError('The number of samples in mat needs' +
+                         'to be the same as the number of samples' +
+                         'in cats')
+    mat = _check_composition(matrix, ignore_zeros=False)
+    cats = pd.Series(categories)
+
+    mat.sort_index(inplace=True)
+    cats = cats.sort_index()
+    labs = mat.columns
+
+    n_samp, n_feat = mat.shape
+
+    _logratio_mat = _log_compare(mat.values, cats.values, significance_test)
+    logratio_mat = _logratio_mat + _logratio_mat.T
+
+    # Multiple comparisons
+    if multiple_comparisons_correction is not None:
+        if multiple_comparisons_correction == 'holm-bonferroni':
+            logratio_mat = np.apply_along_axis(_holm_bonferroni,
+                                               1, logratio_mat)
+
```

Comment:
this will silently fall through if `multiple_comparisons_correction` is not None, but if it is also `!=` to `holm-bonferroni`. possible to add a `raises` in the event that a multiple comparisons correction is specified but is unknown? it would be good to sanity check this early on the assumption that the `_log_compare` is reasonably expensive